src/quick-lint-js/parse-statement-inl.h

```diff
@@ -1593,12 +1595,18 @@ void parser::parse_and_visit_switch(Visitor &v) {
     }
 
     case token_type::kw_default:
+      is_before_first_switch_case = false;
       this->skip();
       QLJS_PARSER_UNIMPLEMENTED_IF_NOT_TOKEN(token_type::colon);
       this->skip();
       break;
 
     default: {
+      if (is_before_first_switch_case) {
+        this->error_reporter_->report(error_statement_before_first_switch_case{
+            .unexpected_statement = this->peek().span(),
```

Comment:
Shouldn't we highlight the entire statement, not just its first token?

Reply:
I think the first token makes sense for example if someone writes this:
```javascript
switch(cond) {
  csae 1: // 'case' misspelled 
    break;
}
```